appsembler/configuration
# Prevent XBlock 1.3.1 from being installed arbitrarily by EDXAPP_EXTRA_REQUIREMENTS

### :warning: This a big change to how we modify our requirements :warning: 


### Background
We found that `EDXAPP_EXTRA_REQUIREMENTS` is installing the latest XBlock all the time and a couple of days ago this broke our Tahoe Staging and would also break Tahoe Production if we don't fix it.

We pushed a hack https://github.com/appsembler/edx-configs/pull/854 to force ping the XBlock to 1.2.2 but it's just a hack. 

I think the correct solution is to either:

 - **Easy:** Merge this PR.
 - **Hard:** Bake all requirements into the `edx-platform` itself and use `pip-tools` to help us find and fix conflicting requirements. While `pip-tools` have been the standard way to mange requirements for Open edX, we never used it as it's mostly impossible to use for a fork that's behind the upstream `master`. Long story here. So I don't recommend this route.

### The Fix

 - **Previously:** Any requirement in `EDXAPP_EXTRA_REQUIREMENTS` gets to install all its dependencies without restrictions. This was often helpful to avoid listing too much dependencies within `EDXAPP_EXTRA_REQUIREMENTS`. 

 - **Proposed:** `EDXAPP_EXTRA_REQUIREMENTS` packages are no longer able to install extra requirements which could break Open edX for new installations. We need to explicitly specifiy the required dependencies within `EDXAPP_EXTRA_REQUIREMENTS` in addition to their own dependencies.

### How to Test this PR
 - For now: Push it to production with https://github.com/appsembler/edx-configs/pull/854 reverted. It should fix the issue of XBlock.
 - Pretty soon: Create a fresh `edxapp` server and deploy, it'll probably break and we need to fix it.
